### ISSUE TITLE
Enhance Random Tile Generator Algorithm

### ISSUE THREAD

**Issue by steven-pereira** [steven-pereira/2048-game#4]
There's an edge case that's causing the random tile generator to fail.

**Comment by jameshuber15** [comment]
I think it's going forever since you have the || case in the for loop so once uniqueTileSet is set to true the loop will be infinite.

**Comment by jameshuber15** [comment]
I think you meant to use break instead of continue.

**Comment by steven-pereira** [comment]
Yeah it goes forever if it never finds an open spot for sure, I actually just added a isGameBoardFull check prior to trying to adding new random random tiles.

The continue looks like leftover from a previous refactor. It's not meant as a break cause i do want to keep trying to find an empty spot.

This implementation in general was a quick one at the time. I'll clean it up :)

**Comment by steven-pereira** [comment]
Love this by the way! :)
